### PR TITLE
Puppet module for distributing CRL

### PR DESCRIPTION
**Pull request by mattj-io**
Consists of -

A generic function to return the contents of a file, only executed on compile time on the puppetmaster
File resource to use the function
README and LICENSE

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,59 @@
...
+
+Do not declare this class on your puppet master, it just gets declared on the host you want the CRL to be copied to. 
+
+In a standard puppet environment where you just want to copy the CRL to the standard puppet location :
+
+    include ::puppetcrl_sync
+
+All the parameters are configurable, so to customise you'd do something like :
+
+    class { '::puppetcrl_sync':
```
Comment (spjmurray) [inline README.md:38]: Up to you but you can format these code blocks as syntax-highlighted puppet code with <backtick><backtick><backtick>puppet

File: manifests/init.pp [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+# [*crl_owner*]
+#   String. Owner for the CRL file in the target location
+# [*crl_group*]
+#   String. Group for the CRL file in the target location
+# [*crl_mode*]
+#   Integer. Octal mode for the CRL file in the target location
+#
+#
+class puppetcrl_sync (
+  $crl_location = $puppetcrl_sync::params::crl_location,
```
Comment (spjmurray) [inline manifests/init.pp]: Again up to you but prefixing everything with crl seems a bit superfluous as there are no name space clashes.

File: manifests/init.pp [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  $crl_group    = $puppetcrl_sync::params::crl_group,
+  $crl_mode     = $puppetcrl_sync::params::crl_mode,
+) inherits ::puppetcrl_sync::params
+{
+
+  validate_string($crl_location)
+  validate_string($crl_target)
+  validate_string($crl_owner)
+  validate_string($crl_group)
+  validate_integer($crl_mode)
```
Comment (spjmurray) [inline manifests/init.pp]: this is a string not an integer

**Comment by yankcrime** [comment]
Naming / style preferences aside, and assuming it works as intended - LGTM 👍

**Pull request merged**